src/useIsShake.ts

```diff
@@ -51,14 +54,41 @@ export function useIsShake() {
     setSubscription(null);
   };
 
+  const checkAccelerometerAvail = async () => {
+    const availability = await isAvailableAsync();
+    setIsAccelerometerAvail(() => {
+      setIsAccelerometerAvailPending(false);
+      return availability;
+    });
+  };
+
+  const getAccelerometerPermiss = async () => {
+    const permissionResponse = await requestPermissionsAsync();
+    if (permissionResponse.granted) setIsAccelerometerAvail(() => {
+      setIsAccelerometerAvailPending(false);
+      return permissionResponse.granted;
```

Comment:
If accelerometer is not already available to the app, a request is made to end user to enable app access to accelerometer.